### PR TITLE
Feature/frans lukas/2241 settings to fragment

### LINKED ISSUE #2241: Convert settings activities to fragments
**Is your feature request related to a problem? Please describe.**

#2235 

Custom and core now uses the navigation component which should make this change simpler than the history/bookmarks activity->fragment ticket.

### PR DESCRIPTION
**Pull request by Frans-Lukas**
Fixes #2241

Settings is now a fragment and is reachable via the navigation component from custom and kiwix. Some small fixes remain such as the tabs not closing when clearing history.

### CODE REVIEW

**Comment by Frans-Lukas** [comment]
```
@Override
  public void onBackPressed() {
    getWindow().setWindowAnimations(0);
    if (allHistoryCleared) {
      Intent data = new Intent();
      data.putExtra(EXTRA_WEBVIEWS_LIST, allHistoryCleared);
      setResult(RESULT_HISTORY_CLEARED, data);
    }
    super.onBackPressed();
  }
```
CoreReaderFragment:
```
....
       if (resultCode == RESULT_HISTORY_CLEARED) {
          webViewList.clear();
          newMainPageTab();
          tabsAdapter.notifyDataSetChanged();
        }
.....

```

@macgills Is this necessary to add back in? I don't think I agree that clearing the webViewList is an expected side effect of clearing one's history? Does it work that way in chrome?

**Comment by macgills** [comment]
Can you check what Chrome does? I think the intent is the clearing of the webBackForward List as a user who has deleted their history on return to the reader would see back/forward buttons that allow them to navigate a history they have just deleted

**Comment by Frans-Lukas** [comment]
Chrome seems to only clear the history. On return to the browser after history clearing, everything is as before the history is cleared. That is, all tabs and their states remain.

**Comment by macgills** [comment]
Then I am fine with it being removed. Can you target the branch this originated from? Reviewing it is a bit difficult

File: core/src/main/java/org/kiwix/kiwixmobile/core/page/PageFragment.kt [hunk 1]
```diff
@@ -96,13 +96,13 @@ abstract class PageFragment : OnItemClickListener, BaseFragment(), FragmentActiv
   }
 
   override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+    super<BaseFragment>.onCreateOptionsMenu(menu, inflater)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/page/PageFragment.kt:99]: I don't think this file was meant to be changed?

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CoreSettingsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.BaseFragment
+
+abstract class CoreSettingsFragment : BaseFragment() {
+
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
+    super.onActivityCreated(savedInstanceState)
+    requireActivity().supportFragmentManager
+      .beginTransaction().replace(R.id.content_frame, createPreferenceFragment()!!)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/CoreSettingsFragment.kt:35]: is the `!!` necessary?

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CoreSettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.BaseFragment
+
+abstract class CoreSettingsFragment : BaseFragment() {
+
+  override fun onActivityCreated(savedInstanceState: Bundle?) {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/CoreSettingsFragment.kt:32]: maybe `OnViewCreated`?

File: custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomMainActivity.kt [hunk 4]
```diff
@@ -79,8 +76,9 @@ class CustomMainActivity : CoreMainActivity() {
       .isVisible = false
   }
 
-  override fun openSettingsActivity() {
-    startActivityForResult(intent<CustomSettingsActivity>(), REQUEST_PREFERENCES)
+  override fun openSettings() {
+    super.openSettings()
+    navigate(R.id.customSettingsFragment)
```
Comment (macgills) [inline custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomMainActivity.kt:81]: why are we not doing this how the others are done? with abstract vals and the implementation of the navigation in Core?

Comment (Frans-Lukas) [reply custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomMainActivity.kt:81]: Good question!

**Pull request merged**